pkg/services/pluginsintegration/pluginexternal/check.go

```diff
@@ -0,0 +1,43 @@
+package pluginexternal
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/plugins/log"
+	"github.com/grafana/grafana/pkg/services/pluginsintegration/pluginstore"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+type Service struct {
+	cfg         *setting.Cfg
+	logger      log.Logger
+	pluginStore pluginstore.Store
+}
+
+func ProvideService(
+	cfg *setting.Cfg, pluginStore pluginstore.Store,
+) (*Service, error) {
+	dslogger := log.New("datasources")
```

Comment:
Nit: This could probably be something like 
```suggestion
	logger := log.New("pluginexternal.validation")
```